android/src/main/java/com/mapbox/mapboxgl/Convert.java

```diff
@@ -301,12 +306,14 @@ static void interpretMapboxMapOptions(Object o, MapboxMapOptionsSink sink) {
     final Object compassViewMargins = data.get("compassViewMargins");
     if(compassViewMargins != null){
       final List compassViewMarginsData = toList(compassViewMargins);
-      sink.setCompassViewMargins(toInt(compassViewMarginsData.get(0)), toInt(compassViewMarginsData.get(1)));
+      final Point point = toPoint(compassViewMarginsData, metrics.density);
+      sink.setCompassViewMargins(point.x, point.y);
     }
     final Object attributionButtonMargins = data.get("attributionButtonMargins");
     if(attributionButtonMargins != null){
       final List attributionButtonMarginsData = toList(attributionButtonMargins);
-      sink.setAttributionButtonMargins(toInt(attributionButtonMarginsData.get(0)), toInt(attributionButtonMarginsData.get(1)));
+      final Point point = toPoint(attributionButtonMarginsData, metrics.density);
+      sink.setAttributionButtonMargins(metrics.widthPixels - point.x - 56, point.y);
```

Comment:
where is the 56 coming from? Maybe give it a name?

Reply:
:+1: on naming constant values. 
Additionally `56` is a fixed pixel value and to support devices with different screensizes/pixelratios, we should take the density in account. Could you check the density of your device and update the value accordingly? (eg. density 2, so  `metrics.widthPixels - point.x - 56` becomes `metrics.widthPixels - point.x - (28 * metrics.density)`

Reply:
Thats the width for the attribution button, I named it